dm/ui/src/components/BatchImportTaskConfig/index.tsx

```diff
@@ -0,0 +1,250 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { clamp } from 'lodash'
+import { useSafeState } from 'ahooks'
+import QueueAnim from 'rc-queue-anim'
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter'
+import { ghcolors } from 'react-syntax-highlighter/dist/esm/styles/prism'
+
+import { Button, Radio, Checkbox, Steps, Alert, StepProps } from '~/uikit'
+import {
+  BatchImportTaskConfigResponse,
+  useDmapiBatchImportTaskConfigMutation,
+} from '~/models/taskConfig'
+import { sleep } from '~/utils/sleep'
+
+const { Step } = Steps
+
+const BlinkingText: React.FC<{ text: string; active: boolean }> = ({
+  text,
+  active,
+}) => {
+  const [count, setCount] = useSafeState(1)
+  let timer = useRef<number>()
+
+  useEffect(() => {
+    if (active) {
+      timer.current = window.setInterval(() => setCount(c => c + 1), 300)
+    }
+    return () => {
+      if (timer.current) {
+        clearInterval(timer.current)
+      }
+    }
+  }, [active])
+
+  if (!active) {
+    return <span>{text}</span>
+  }
+  return <span>{text + '.'.repeat(count % 4)}</span>
+}
+
+const BatchImportTaskConfig: React.FC = () => {
+  const [t] = useTranslation()
+  const [percent, setPercent] = useSafeState(0)
+  const [error, setError] = useState<any>()
+  const [importResult, setImportResult] =
+    useState<BatchImportTaskConfigResponse>()
```

Comment:
Curious won't it warn if the useState parameter is empty?

Reply:
`importResult` type is actually `BatchImportTaskConfigResponse | undefined`, when access it it'll need optional chaining